android/src/test/java/com/amplitude/android/AmplitudeSessionTest.kt

```diff
@@ -604,6 +632,72 @@ class AmplitudeSessionTest {
         Assertions.assertEquals(1, tracks.count())
     }
 
+    @Test
+    fun amplitude_shouldStartNewSessionOnInitializationInForegroundBasedOnSessionTimeout() = runTest {
+        val startTime: Long = 1000
+        var time: Long = startTime
+
+        mockSystemTime(time)
+
+        val storageProvider = InstanceStorageProvider(InMemoryStorage())
+        val config = createConfiguration(storageProvider)
+
+        // Create an instance in the background
+        val amplitude1 = Amplitude(config)
+        setDispatcher(amplitude1, testScheduler)
+        amplitude1.isBuilt.await()
+
+        // enter foreground (will start a session)
+        amplitude1.onEnterForeground(time)
```

Comment:
one edge case I can think of is when it is initialized first, then create an event before onEnterForeground is triggered. Curious if that's possible or already handled

Reply:
Thanks @qingzhuozhen I believe that should be already handled. See the background/foreground tests.

In the case of a init in BG, track event, enter FG, while in BG sessions are subject to `minTimeBetweenSessionsMillis`, so if events are more than that apart a new session will be started.